File: app/src/main/java/com/innerly/app/ui/ReflectionAdapter.java
package com.innerly.app.ui;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import com.innerly.app.R;
import com.innerly.app.data.Reflection;
import java.text.SimpleDateFormat;
import java.util.ArrayList;
import java.util.Date;
import java.util.List;
import java.util.Locale;

public class ReflectionAdapter extends RecyclerView.Adapter<ReflectionAdapter.ReflectionViewHolder> {

    private List<Reflection> reflections = new ArrayList<>();

    // දත්ත අලුත් කිරීම සඳහා method එක
    public void setReflections(List<Reflection> reflections) {
        this.reflections = reflections;
        notifyDataSetChanged();
    }

    @NonNull
    @Override
    public ReflectionViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        // item_reflection layout එක inflate කිරීම
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_reflection, parent, false);
        return new ReflectionViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ReflectionViewHolder holder, int position) {
        // අදාළ ස්ථානයේ ඇති Reflection එක ලබාගෙන ViewHolder එකට සම්බන්ධ කිරීම
        Reflection reflection = reflections.get(position);
        holder.bind(reflection);
    }

    @Override
    public int getItemCount() {
        return reflections.size();
    }

    // ViewHolder Class එක
    public static class ReflectionViewHolder extends RecyclerView.ViewHolder {
        private final TextView tvDate;
        private final TextView tvMood;
        private final TextView tvPreview;

        public ReflectionViewHolder(@NonNull View itemView) {
            super(itemView);
            tvDate = itemView.findViewById(R.id.tvReflectionDate);
            tvMood = itemView.findViewById(R.id.tvReflectionMood);
            tvPreview = itemView.findViewById(R.id.tvReflectionPreview);
        }

        public void bind(Reflection reflection) {
            // දිනය Format කිරීම (MMM dd, yyyy)
            SimpleDateFormat sdf = new SimpleDateFormat("MMM dd, yyyy", Locale.getDefault());
            String formattedDate = sdf.format(new Date(reflection.getDate()));

            tvDate.setText(formattedDate);
            tvMood.setText(reflection.getMood());
            tvPreview.setText(reflection.getText());
        }
    }
}

File: app/src/main/java/com/innerly/app/ui/ReflectionAdapter.kt
package com.innerly.app.ui

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.innerly.app.R
import com.innerly.app.data.Reflection
import java.text.SimpleDateFormat
import java.util.Date
import java.util.Locale

class ReflectionAdapter : RecyclerView.Adapter<ReflectionAdapter.ReflectionViewHolder>() {

    private var reflections: List<Reflection> = emptyList()

    fun setReflections(reflections: List<Reflection>) {
        this.reflections = reflections
        notifyDataSetChanged()
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ReflectionViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.item_reflection, parent, false)
        return ReflectionViewHolder(view)
    }

    override fun onBindViewHolder(holder: ReflectionViewHolder, position: Int) {
        holder.bind(reflections[position])
    }

    override fun getItemCount(): Int = reflections.size

    class ReflectionViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        private val tvDate: TextView = itemView.findViewById(R.id.tvReflectionDate)
        private val tvMood: TextView = itemView.findViewById(R.id.tvReflectionMood)
        private val tvPreview: TextView = itemView.findViewById(R.id.tvReflectionPreview)

        fun bind(reflection: Reflection) {
            val sdf = SimpleDateFormat("MMM dd, yyyy", Locale.getDefault())
            tvDate.text = sdf.format(Date(reflection.date))
            tvMood.text = reflection.mood
            tvPreview.text = reflection.text
        }
    }
}